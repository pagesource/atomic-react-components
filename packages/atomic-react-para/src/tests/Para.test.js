import React from 'react';
import { shallow, render } from 'enzyme';

import { ParaVanilla } from '..';

const props = {
  children: 'Sample',
  className: '',
  elementLocator: 'my-paragraph',
};

describe('Para Component', () => {
  let paragraph;

  test('should render correctly and match snapshot', () => {
    paragraph = shallow(<ParaVanilla {...props}>This is Para text</ParaVanilla>);
    expect(paragraph).toMatchSnapshot();
  });
  test('should check the data attribute of para', () => {
    const para = render(<ParaVanilla {...props} />);

    expect(para.attr('data-locator')).toEqual(`para-${props.elementLocator}`);
  });
});
