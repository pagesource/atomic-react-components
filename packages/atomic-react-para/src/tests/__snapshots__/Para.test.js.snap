// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Para Component should render correctly and match snapshot 1`] = `
<p
  className=""
  data-locator="para-my-paragraph"
>
  This is Para text
</p>
`;
